examples/cacoh.py

```diff
@@ -0,0 +1,496 @@
+# :footcite:`EwaldEtAl2012`).
+
+# %%
+
+# Compute CaCoh following rank subspace projection
+cacoh_red = spectral_connectivity_epochs(
+    data,
+    method="cacoh",
+    indices=multivar_indices,
+    sfreq=100,
+    fmin=3,
+    fmax=35,
+    rank=([2], [2]),
+)
+
+# compare standard and rank subspace-projected CaCoh
+fig, axis = plt.subplots(1, 1)
+axis.plot(cacoh.freqs, np.abs(cacoh.get_data()[0]), linewidth=2, label="standard CaCoh")
+axis.plot(
+    cacoh_red.freqs,
+    np.abs(cacoh_red.get_data()[0]),
+    linewidth=2,
+    label="rank subspace (2) CaCoh",
+)
+axis.set_xlabel("Frequency (Hz)")
+axis.set_ylabel("Connectivity (A.U.)")
+axis.legend(loc="lower right")
+fig.suptitle("CaCoh")
+
+# no. channels equal with and without projecting to rank subspace for patterns
+assert patterns[0, 0].shape[0] == np.array(cacoh_red.attrs["patterns"])[0, 0].shape[0]
+assert patterns[1, 0].shape[0] == np.array(cacoh_red.attrs["patterns"])[1, 0].shape[0]
+
+###############################################################################
+# See :doc:`mic_mim` for an example of applying the rank subspace
+# projection to real data with a large number of channels.
+#
+# In the case that your data is not full rank and ``rank`` is left as ``None``,
+# an automatic rank computation is performed and an appropriate degree of
+# dimensionality reduction will be enforced. The rank of the data is determined
+# by computing the singular values of the data and finding those within a
+# factor of :math:`1e^{-6}` relative to the largest singular value.
+#
+# Whilst unlikely, there may be scenarios in which this threshold may be too
+# lenient. In these cases, you should inspect the singular values of your data
+# to identify an appropriate degree of dimensionality reduction to perform,
+# which you can then specify manually using the ``rank`` argument. The code
+# below shows one possible approach for finding an appropriate rank of
+# close-to-singular data with a more conservative threshold.
+
+# %%
+
+# gets the singular values of the data across epochs
+s = np.linalg.svd(data, compute_uv=False).min(axis=0)
+# finds how many singular values are 'close' to the largest singular value
+rank = np.count_nonzero(s >= s[0] * 1e-4)  # 1e-4 is the 'closeness' criteria
+
+###############################################################################
+# Limitations
+# -----------
+#
+# Multivariate methods offer many benefits in the form of dimensionality
+# reduction and signal-to-noise ratio improvements, however no method is
+# perfect. When we simulated the data, we mentioned how we considered the seeds
+# and targets to be signals of different modalities. This is an important
+# factor in whether CaCoh should be used over methods based solely on the
+# imaginary part of coherency such as MIC and MIM.
+#
+# In short, if you want to examine connectivity between signals from the same
+# modality, you should consider not using CaCoh. Instead, methods based on the

```

Comment:
```suggestion
# In short, if you want to examine connectivity between signals from the same
# modality, you should consider using another method instead of CaCoh. Instead, methods based on the

```

Reply:
Done:
https://github.com/mne-tools/mne-connectivity/blob/1022e83d133bf4b860d067f855d9147ccb07168c/examples/cacoh.py#L473-L475